File: javascript/services/NewsPlatform.js
const BitcoinPriceObserver = require('./BitcoinPriceObserver');

class NewsPlatform extends BitcoinPriceObserver {
    update(oldPrice, newPrice) {
        console.log('\n\nNewsPlatform');
        console.log(`Price updated: ${oldPrice} -> ${newPrice}`);
    }
}

module.exports = NewsPlatform;

File: src/main/java/com/douglasmatosdev/service/NewsPlatform.java
package com.douglasmatosdev.service;

public class NewsPlatform implements BitcoinPriceObserver {

    @Override
    public void update(Double oldPrice, Double newPrice) {
        if (newPrice > oldPrice) {
            System.out.println("Price increased! Notify news platform!");
        } else {
            System.out.println("Price decreased! Notify news platform!");
        }
    }
}
